src/common/QuirksManager.cpp

```diff
@@ -135,12 +139,42 @@ class SamsungDeviceQuirks : public  QuirksManager::DeviceQuirks {
     int mBuildChangelist = 0;
 };
 
+class QualcommDeviceQuirks : public  QuirksManager::DeviceQuirks {
+public:
+    QualcommDeviceQuirks() {
+        std::string modelName = getPropertyString("ro.soc.model");
+        isSM8150 = (modelName == "SDM8150");
+    }
+
+    virtual ~QualcommDeviceQuirks() = default;
+
+    bool isMMapSafe(const AudioStreamBuilder &builder) override {
+        // See https://github.com/google/oboe/issues/1121#issuecomment-897957749
+        bool mmapBroken = false;
+        if (isSM8150 && (getSdkVersion() <= __ANDROID_API_P__)) {
+            LOGI("QuirksManager::%s() MMAP not actually supported on this chip."
+                 " Switching off MMAP.", __func__);
+            mmapBroken = true;
+        }
+
+        return !mmapBroken;
+    }
+
+private:
+    bool isSM8150 = false;
+};
+
 QuirksManager::QuirksManager() {
-    std::string manufacturer = getPropertyString("ro.product.manufacturer");
-    if (manufacturer == "samsung") {
+    std::string productManufacturer = getPropertyString("ro.product.manufacturer");
+    if (productManufacturer == "samsung") {
         mDeviceQuirks = std::make_unique<SamsungDeviceQuirks>();
     } else {
-        mDeviceQuirks = std::make_unique<DeviceQuirks>();
+        std::string socManufacturer = getPropertyString("ro.soc.manufacturer");
```

Comment:
This code makes it seem that a device will is either a Samsung Device or a Qualcomm device. Would it be possible for devices to have multiple Quirks in the future?

We should worry about this later but I think the next time we add a new Quirk, like let's say for Pixel 4a, we should consider doing some refactor of this.

Reply:
Good Point. We should distinguish more clearly between the Samsung Exynos quirks and the SamsungMobile quirks. They are quite different.